Samples/SqueezeNetObjectDetection/NETCore/cs/Program.cs

```diff
@@ -117,17 +120,53 @@ private static T AsyncHelper<T> (IAsyncOperation<T> operation)
             return operation.GetResults();
         }
 
-        private static ImageFeatureValue LoadImageFile()
+        private static ImageFeatureValue LoadImageFile(ColorManagementMode colorManagementMode)
         {
             StorageFile imageFile = AsyncHelper(StorageFile.GetFileFromPathAsync(_imagePath));
             IRandomAccessStream stream = AsyncHelper(imageFile.OpenReadAsync());
             BitmapDecoder decoder = AsyncHelper(BitmapDecoder.CreateAsync(stream));
-            SoftwareBitmap softwareBitmap = AsyncHelper(decoder.GetSoftwareBitmapAsync());
+            SoftwareBitmap softwareBitmap = null;
+            try
+            {
+                softwareBitmap = AsyncHelper(
+                    decoder.GetSoftwareBitmapAsync(
+                        decoder.BitmapPixelFormat,
+                        decoder.BitmapAlphaMode,
+                        new BitmapTransform(),
+                        ExifOrientationMode.RespectExifOrientation,
+                        colorManagementMode
+                    )
+                );
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Failed to create SoftwareBitmap! Please make sure that input image is within the model's colorspace.");
+                Console.WriteLine(" Exception caught.\n {0}", e);
+                System.Environment.Exit(e.HResult);
+            }
             softwareBitmap = SoftwareBitmap.Convert(softwareBitmap, BitmapPixelFormat.Bgra8, BitmapAlphaMode.Premultiplied);
             VideoFrame inputImage = VideoFrame.CreateWithSoftwareBitmap(softwareBitmap);
             return ImageFeatureValue.CreateFromVideoFrame(inputImage);
         }
 
+        private static ColorManagementMode GetColorManagementMode()
+        {
+            // Get model color space gamma
+            string gammaSpace = "";
+            bool doesModelContainGammaSpaceMetadata = _model.Metadata.TryGetValue("Image.ColorSpaceGamma", out gammaSpace);

```

Comment:
Out of curiosity is there documentation around the meaning of "Image.ColorSpaceGamma"? I couldn't find it online after a quick search.

Reply:
Yep! https://github.com/onnx/onnx/blob/master/docs/MetadataProps.md